contracts/Hub.sol

```diff
@@ -107,71 +91,33 @@ contract Hub {
 
     // Trust does not have to be reciprocated.
     // (e.g. I can trust you but you don't have to trust me)
-    function trust(address toTrust, bool yes, uint limit) public {
+    function trust(address toTrust, uint limit) public {
         require(trustable(toTrust));
-        edges[msg.sender][toTrust] = yes ? EdgeWeight(limit, 0, time()) : EdgeWeight(0, 0, 0);
+        limits[msg.sender][toTrust] = limit;
         emit Trust(msg.sender, toTrust, limit);
     }
 
-    function updateTrustLimit(address toUpdate, uint256 limit) public {
-        require(trustable(toUpdate));
-        require(address(tokenToUser[toUpdate]) != address(0));
-        edges[msg.sender][toUpdate] = EdgeWeight(limit, 0, time());
-        emit UpdateTrustLimit(msg.sender, toUpdate, limit);
+    function checkSendLimit(address from, address to) public view returns (uint256) {
+        uint256 max = (userToToken[to].totalSupply().mul(limits[to][from])).div(100);
+        return max.sub(userToToken[to].balanceOf(from));
     }
 
     // Starts with msg.sender then ,
     // iterates through the nodes list swapping the nth token for the n+1 token
-    function transferThrough(address[] memory nodes, address[] memory tokens, uint wad) public {
-
-        uint tokenIndex = 0;
-
-        address prevValidator;
+    function transferThrough(address[] memory users, uint wad) public {
+        require(users.length <= 5);
+        address prev = msg.sender;
+        for (uint i = 0; i < users.length; i++) {
+            address curr = users[i];
+            uint256 max = (userToToken[curr].totalSupply().mul(limits[curr][prev])).div(1000000000000000000);
```

Comment:
Does the division by `1000000000000000000` exist to convert the denominator? Numbers like this I would move to a named constant to make it more readable / understandable.